### PR TITLE
Fix pg10 issues

### PR DESCRIPTION
**Pull request by ethervoid**
Related #234 

The main problem is the different behavior of the `unnest` functions between PG 9.6 and 10 versions

### CODE REVIEW

**Comment by skgsergio** [comment]
retest this please

**Comment by skgsergio** [comment]
I've ran this PR in the new CI with PG 10: https://ci.cartodb.net/job/CartoDB-DATASERVICES-PR-testing/14/console

File: geocoder/extension/sql/0.0.1/20_admin0.sql [hunk 1]
```diff
@@ -8,11 +8,12 @@ CREATE OR REPLACE FUNCTION geocode_admin0_polygons(name text[])
...
   -- FOR ret IN
   RETURN QUERY
-    SELECT d.q, n.the_geom as geom, CASE WHEN s.adm0_a3 IS NULL then FALSE ELSE TRUE END AS success
-      FROM (SELECT q, lower(geocode_clean_name(q))::text x
-        FROM (SELECT unnest(name) q) g) d
-      LEFT OUTER JOIN admin0_synonyms s ON name_ = d.x
-      LEFT OUTER JOIN ne_admin0_v3 n ON s.adm0_a3 = n.adm0_a3 GROUP BY d.q, n.the_geom, s.adm0_a3;
+    SELECT q, n.the_geom as geom, CASE WHEN s.adm0_a3 IS NULL then FALSE ELSE TRUE END AS success
+    FROM unnest(name) WITH ORDINALITY q
+    LEFT OUTER JOIN admin0_synonyms s ON name_ = lower(geocode_clean_name(q))::text
```
Comment (juanignaciosl) [inline geocoder/extension/sql/0.0.1/20_admin0.sql]: Much more readable ❤️

File: geocoder/extension/sql/0.0.1/50_namedplaces.sql [hunk 2]
```diff
@@ -260,9 +260,18 @@ CREATE OR REPLACE FUNCTION geocode_namedplace(places text[], admin1s text[], inp
           RETURN NEXT ret;
       END LOOP;
     ELSE
-      FOR ret IN WITH clean AS (SELECT array_agg(p) p, array_agg(a) a, c FROM (SELECT p, a, c FROM (SELECT unnest(places) p, unnest(admin1s) a, unnest(inputcountry) c) z GROUP BY p, a, c) y GROUP BY c)
-        SELECT (geocode_namedplace(p, a, c)).* FROM clean LOOP
-          RETURN NEXT ret;
+      FOR ret IN WITH clean AS (
+          SELECT array_agg(p) p, array_agg(a) a, c
+          FROM (SELECT p, a, c
+                FROM (SELECT p, a, c, nest.ordinality as ord FROM unnest(places, admin1s) with ordinality nest (p, a), LATERAL unnest(inputcountry) with ordinality c) z
```
Comment (juanignaciosl) [inline geocoder/extension/sql/0.0.1/50_namedplaces.sql]: I didn't know about the new `unnest` with multiple arguments ❤️

Comment (juanignaciosl) [reply geocoder/extension/sql/0.0.1/50_namedplaces.sql]: Is LATERAL actually needed?

Comment (ethervoid) [reply geocoder/extension/sql/0.0.1/50_namedplaces.sql]: Yes, we need to fill the country column with the value for all the rows. That was done by unnest in previous versions but it fails in pg10

**Pull request merged**